### PR TITLE
Bug 1148791 - [Build] Cache the result of downloading modules.tar r=rick...

### PR DESCRIPTION
**Pull request by julienw**
...ychien

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=78ba43e8cff45ad9561615b598b4b7d3951de5e6

File: Makefile [hunk 1]
```diff
@@ -702,10 +703,13 @@ $(NPM_INSTALLED_PROGRAMS): package.json node_modules
...
-modules.tar: gaia_node_modules.revision
+modules.tar: gaia_node_modules.revision $(NODE_MODULES_CACHEDIR)/$(NODE_MODULES_REV)
+	-cp -f "$(NODE_MODULES_CACHEDIR)/$(NODE_MODULES_REV)" "$(NODE_MODULES_SRC)"
+
+$(NODE_MODULES_CACHEDIR)/$(NODE_MODULES_REV): gaia_node_modules.revision
 	@echo Downloading latest node_modules package. This may take several minutes...
-	-$(DOWNLOAD_CMD) https://github.com/mozilla-b2g/gaia-node-modules/tarball/$(NODE_MODULES_REV) &&\
-	mv $(NODE_MODULES_REV) "$(NODE_MODULES_SRC)"
+	mkdir -p "$(NODE_MODULES_CACHEDIR)"
+	-cd "$(NODE_MODULES_CACHEDIR)" && $(DOWNLOAD_CMD) https://github.com/mozilla-b2g/gaia-node-modules/tarball/$(NODE_MODULES_REV)
```
Comment (julienw) [inline Makefile]: using "-" ensures we can run this even when our network is failing

File: Makefile [hunk 2]
```diff
@@ -98,6 +98,7 @@ endif
 
 DOGFOOD?=0
 NODE_MODULES_SRC?=modules.tar
+NODE_MODULES_CACHEDIR=modules.tar.cachedir
```
Comment (RickyChien) [inline Makefile]: nit: I prefer to name all our folder name with _ instead of .
modules_tar_cachedir

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=43dc9e072ad16a89664e70f161a312e6acb35c43

**Pull request merged**